nncf/common/pruning/utils.py

```diff
@@ -327,10 +321,14 @@ def count_flops_and_weights_per_node(graph: NNCFGraph,
         name = node.node_name
         if name in op_addresses_to_skip:
             continue
-        flops_numpy = 2 * np.prod(input_shapes[name]) * np.prod(output_shapes[name])
-        weights_numpy = np.prod(input_shapes[name]) * np.prod(output_shapes[name])
-        flops[name] = flops_numpy.astype(int).item()
-        weights[name] = weights_numpy.astype(int).item()
+
+        num_in_features = node.layer_attributes.in_features
+        num_out_features = node.layer_attributes.out_features
+
+        flops_numpy = 2 * num_in_features * num_out_features
+        weights_numpy = num_in_features * num_out_features
```

Comment:
Is matrix multiplication complexity `O(mnk)` for `(m, n)` and `(n, k)` matrices? I think that we should consider `input_shapes(m)` in addition to `num_in_features(n)` and `num_in_features(k)`. Am I right?

Reply:
Like this https://github.com/sovrasov/flops-counter.pytorch/blob/06793e8ec1b73b154b8078db571ac682125f73ce/ptflops/pytorch_ops.py#L31-L36

Reply:
Agree, this will be fixed in #1210